miniapp/io.cpp

```diff
@@ -280,7 +276,6 @@ cl_options read_options(int argc, char** argv, bool allow_write) {
         update_option(options.bin_regular, bin_regular_arg);
         update_option(options.all_to_all, all_to_all_arg);
         update_option(options.ring, ring_arg);
-        update_option(options.group_size, group_size_arg);
```

Comment:
Adding support for user control over group sizes, and other parameters, would require some interface for passing policies/hints to the load-balancer/domain decomposition. This isn't too complex, but for the sake of simplicity I didn't include that in this set of changes.
We can do this in a follow up task?